src/DynamoCoreWpf/Controls/DynamoNodeButton.cs

```diff
@@ -54,16 +59,16 @@ public DynamoNodeButton(ModelBase model, string eventName)
 
         private void OnDynamoNodeButtonClick(object sender, RoutedEventArgs e)
         {
-            // If this DynamoNodeButton was created with an associated model 
-            // and the event name, then the owner of this button (a ModelBase) 
+            // If this DynamoNodeButton was created with an associated model
+            // and the event name, then the owner of this button (a ModelBase)
             // needs the "ModelEventCommand" to be sent when user clicks
             // on the button.
-            // 
+            //
             if (null != this.model && (!string.IsNullOrEmpty(this.eventName)))
             {
                 MessageBoxResult result = MessageBoxResult.None;
 
-                if (eventName.Equals("RemoveInPort"))
+                if (eventName.Equals("RemoveInPort") && ShowWarningForRemovingInPort)
```

Comment:
what is the deal with this? Were you going to set this true in a previous use case or something?

Reply:
> This was from the current Master pull. The `VariableInputNode `from Master now shows a warning dialog box whenver the user click on the minus button to remove a port. For the Watch3D node, we don't need this dialog box to show up, hence I added the `ShowWarningForRemovingInPort `flag to disable it

I'm still confused - I never see you set this true, and the default will be false, so does this affect existing nodes? Did you intend to set it true for existing nodes?

Reply:
Yep you are right. I should have give it a "true" default value. Fixed now